src/Server/grpc_protos/clickhouse_grpc.proto

```diff
@@ -123,7 +121,23 @@ message QueryInfo {
    // zstd: 1..22; 3 is recommended by default (compression level 0 also means 3)
    // lz4: 0..16; values < 0 mean fast acceleration
    // bz2: 1..9
-   int32 compression_level = 19;
+   int32 output_compression_level = 19;
+
+   // Transport compression is an alternative way to make the server to compress its response.
+   // This kind of compression implies that instead of compressing just `output` the server will compress whole packed messages of the `Result` type,
+   // and then gRPC implementation on client side will decompress those messages so client code won't be bothered with decompression.
+   // Here is a big difference between the transport compression and the compression enabled by setting `output_compression_type` because
+   // in case of the transport compression the client code receives already decompressed data in `output`.
+   // If the transport compression is not set here it can still be enabled by the server configuration.
+   // Supported compression types: none, deflate, gzip, stream_gzip
+   // Supported compression levels: 0..3
+   // WARNING: Don't set `transport_compression` and `output_compression` at the same time because it will make the server to compress its output twice!
+   string transport_compression_type = 22;
+   int32 transport_compression_level = 23;
+
+   /// Obsolete fields, should not be used in new code.
+   ObsoleteTransportCompression obsolete_result_compression = 17;
+   string obsolete_compression_type = 18;
```

Comment:
I've renamed these fields by adding the prefix "obsolete_". But their numbers (17 and 18) are still the same and they are still supported in the code of ClickHouse, so any old compiled code will able to work with those fields after that renaming (because compiled code searches fields in a protobuf message by their numbers).